rstcheck/__init__.py

```diff
@@ -516,18 +525,48 @@ def load_configuration_from_file(directory: str, args: argparse.Namespace) -> ar
 
 
 def _get_options(directory_or_file: str, debug: bool = False) -> typing.Dict[str, str]:
+    options = {}
+
     config_path = find_config(directory_or_file, debug=debug)
-    if not config_path:
-        return {}
+    if config_path:
+        config_file = os.path.basename(config_path)
 
-    parser = configparser.ConfigParser()
-    parser.read(config_path)
-    try:
-        options = dict(parser.items("rstcheck"))
-    except configparser.NoSectionError:
-        return {}
+    if config_file == "pyproject.toml":
+        options = _get_pyproject_options(config_path)
     else:
-        return options
+        parser = configparser.ConfigParser()
+        parser.read(config_path)
+        try:
+            options = dict(parser.items("rstcheck"))
+        except configparser.NoSectionError:
+            options = {}
+
+    return options
+
+
+def _get_pyproject_options(config_path: str) -> typing.Dict[str, str]:
```

Comment:
What do you think of this version for this function?
```python
def _get_pyproject_options(config_path: str) -> typing.Dict[str, str]:
    RSTCHECK_TOML_CONFIG = typing.Dict[str, typing.Union[str, typing.List[str]]]

    with open(config_path, "rb") as conf_file:
        config = tomli.load(conf_file)
    
    options_from_file: typing.Optional[RSTCHECK_TOML_CONFIG] = config.get("tool", {}).get("rstcheck", None)

    if options_from_file is None:
        return {}

    options = {}

    # tomli returns a list of strings and ConfigParser returns a comma
    # separated string.  This makes the options from pyproject.toml consistent
    # with the options read from other configuration files.  The try block
    # accounts for pyproject files without a rstcheck section.
    for option in [
        "ignore_directives",
        "ignore_roles",
        "ignore_messages",
        "ignore_language",
        "report",
    ]:
        option_value = options_from_file.get(option, None)
        if option_value is None:
            continue
        
        if isinstance(option_value, list):
            option_value = ",".join(option_value)
        
        options[option] = option_value

    return options
```

Again early returns. I think the readability is good here with less nesting. Also the change of the try/except block to actually check the condition instead of silencing the error makes the intention more clear. This would also be another improvement one could make with my proposal for the try/except block in `_get_options` above.